extensions/security/iprange/src/org/exist/security/realm/iprange/IPRangeServlet.java

```diff
@@ -0,0 +1,121 @@
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.exist.security.AbstractRealm;
+import org.exist.security.AuthenticationException;
+import org.exist.security.SecurityManager;
+import org.exist.security.Subject;
+import org.exist.xquery.XQueryContext;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * IPRange authenticator servlet.
+ *
+ * @author <a href="mailto:wshager@gmail.com">Wouter Hager</a>
+ */
+public class IPRangeServlet extends HttpServlet {
+
+    protected final static Logger LOG = LogManager.getLogger(IPRangeServlet.class);
+    private static final long serialVersionUID = -568037449837549034L;
+    public static AbstractRealm realm = null;
+
+    public IPRangeServlet() throws ServletException {
+    }
+
+    @Override
+    public void init(final ServletConfig config) throws ServletException {
+        super.init(config);
+    }
+
+    @Override
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp)
+            throws ServletException, IOException {
+        doPost(req, resp);
+    }
+
+    @Override
+    protected void doPost(final HttpServletRequest request, final HttpServletResponse response)
+            throws ServletException, IOException {
+
+
+        // Get reverse proxy header when available, otherwise use regular IP address
+        String ipAddress = request.getHeader("X-Forwarded-For");
+        if (ipAddress == null) {
+            ipAddress = request.getRemoteAddr();
+        }
+
+
+        LOG.info("Detected IPaddress " + ipAddress);
+
+        String jsonResponse = "{\"fail\":\"IP range not authenticated\"}";
+
+        try {
+            final SecurityManager securityManager = IPRangeRealm.instance.getSecurityManager();
+            final Subject user = securityManager.authenticate(ipAddress, ipAddress);
+
+            if (user != null) {
+                LOG.info("IPRangeServlet user " + user.getUsername() + " found");
+
+                // Security check
+                if (user.hasDbaRole()) {
+                    LOG.error("User " + user.getUsername() + " has DBA rights, will not be authorized");
+                    return;
+                }
+
+
+                final HttpSession session = request.getSession();
+                // store the user in the session
+                if (session != null) {
+                    jsonResponse = "{\"user\":\"" + user.getUsername() + "\",\"isAdmin\":\"" + user.hasDbaRole() + "\"}";
+                    LOG.info("IPRangeServlet setting session attr " + XQueryContext.HTTP_SESSIONVAR_XMLDB_USER);
+                    session.setAttribute(XQueryContext.HTTP_SESSIONVAR_XMLDB_USER, user);
```

Comment:
this looks a bit wrong. Do you want to authenticate to any operations on server?

If yes then here better [code](https://github.com/eXist-db/exist/blob/develop/extensions/security/oauth/src/org/exist/security/realm/oauth/ServiceFacebook.java#L118-L131) for it.